src/System.Globalization/tests/ISOWeekTests.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.Globalization
+{
+    public static class ISOWeekTests
+    {
+        // The following 71 years in a 400-year cycle have 53 weeks (371 days).
+        // Years not listed have 52 weeks (364 days). Add 2000 for current years.
+        private static readonly HashSet<int> s_YearsWith53Weeks = new HashSet<int>
+        {
+            004, 009, 015, 020, 026,
+            032, 037, 043, 048, 054,
+            060, 065, 071, 076, 082,
+            088, 093, 099, 105, 111, 116, 122,
+            128, 133, 139, 144, 150,
+            156, 161, 167, 172, 178,
+            184, 189, 195, 201, 207, 212, 218,
+            224, 229, 235, 240, 246,
+            252, 257, 263, 268, 274,
+            280, 285, 291, 296, 303, 308, 314,
+            320, 325, 331, 336, 342,
+            348, 353, 359, 364, 370,
+            376, 381, 387, 392, 398
+        };
+
+        // From https://en.wikipedia.org/wiki/ISO_week_date#Relation_with_the_Gregorian_calendar.
+        private static readonly DateData[] s_DateData =
```

Comment:
Nit: Date => date... will stop commenting on these... in general it should be s_camelCase.